src/edu/washington/escience/myria/operator/CSVFileScanFragment.java

```diff
@@ -237,28 +234,30 @@ protected TupleBatch fetchNextReady() throws IOException, DbException {
          * large row that was split among other workers)
          */
         if (onLastRow && !finishedReadingLastRow && !isLastWorker) {
-          long movingEndByte = adjustedEndByteRange;
+          long movingEndByte = adjustedEndByteRange + 1;
           long characterPositionAtBeginningOfRecord = record.getCharacterPosition();
           boolean newLineFound = false;
           while (!newLineFound) {
-            movingEndByte += byteOverlap;
+            // In order to find the new line character, we use SequenceInputStream to pipe in more bytes
             InputStream trailingEndInputStream =
                 source.getInputStream(
-                    adjustedEndByteRange + 1, Math.min(movingEndByte, maxByteRange));
-            int dataChar = trailingEndInputStream.read();
+                    movingEndByte, Math.min(movingEndByte + byteOverlap, maxByteRange));
+            movingEndByte += byteOverlap;
+            partitionInputStream =
```

Comment:
I don't understand the purpose of the `SequenceInputStream` here. If `partitionInputStream` has been exhausted, then why would we concatenate it with a new `InputStream`? Isn't it enough just to scan each `trailingEndInputStream` for newlines?